lib/asciidoctor/chart/docinfo_processor.rb

```diff
@@ -44,7 +29,23 @@ class DocinfoProcessor < ::Asciidoctor::Extensions::DocinfoProcessor
 
       def process doc
         engines = doc.x_chart[:engines]
-        (engines.map {|engine| ASSETS[engine.to_sym] }.to_a + [DEFAULT_STYLE]).join("\n")
+        result = []
+        engines.each do |engine|
+          # noinspection RubyCaseWithoutElseBlockInspection
+          case engine
+          when 'c3js'
+            result << %(<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/c3/0.3.0/c3.min.css">)
+            result << %(<script src="https://cdnjs.cloudflare.com/ajax/libs/d3/3.4.11/d3.min.js" charset="utf-8"></script>)
+            result << %(<script src="https://cdnjs.cloudflare.com/ajax/libs/c3/0.3.0/c3.min.js"></script>)
+          when 'chartjs'
+            result << %(<script src="#{doc.normalize_web_path 'chart.min.js', (doc.attr 'chartjsdir', 'https://cdnjs.cloudflare.com/ajax/libs/Chart.js/3.7.0/'), false}"></script>)
+          when 'chartist'
+            result << %(<link rel="stylesheet" href="https://cdn.jsdelivr.net/chartist.js/latest/chartist.min.css">)
+            result << %(<script src="https://cdn.jsdelivr.net/chartist.js/latest/chartist.min.js"></script>)
+          end
+        end
+        result << [DEFAULT_STYLE]
```

Comment:
I think we don't need to wrap `DEFAULT_STYLE` into an array anymore

Reply:
> I think we don't need to wrap `DEFAULT_STYLE` into an array anymore

Done.

Reply:
Do we need https://cdnjs.cloudflare.com/ajax/libs/d3/3.4.11/d3.min.js for c3 engine? I checked https://cdnjs.com/libraries/c3/0.3.0 and there is no d3 to include.

Reply:
I disabled LineLength check for linting, I hope that is ok.